src/shipit_code_coverage/tests/test_artifacts.py

```diff
@@ -59,3 +60,27 @@ def add_dir(files):
 
     with pytest.raises(Exception, message='suite and chunk can\'t both have a value'):
         a.get(chunk='xpcshell-7', suite='mochitest')
+
+
+@mock.patch('shipit_code_coverage.taskcluster.get_task_artifacts')
+@mock.patch('shipit_code_coverage.taskcluster.download_artifact')
+def test_download(mocked_download_artifact, mocked_get_task, LINUX_TEST_TASK_ARTIFACTS):
+    a = ArtifactsHandler([], [], [])
+    mocked_get_task.return_value = LINUX_TEST_TASK_ARTIFACTS['artifacts']
+    mocked_download_artifact.return_value = LINUX_TEST_TASK_ARTIFACTS['artifacts'][0]
+
+    a.download(LINUX_TEST_TASK_ARTIFACTS)
+
+    assert mocked_download_artifact.call_count == 1
+    assert mocked_get_task.call_count == 1
+    mocked_download_artifact.assert_called_with(
+        [
```

Comment:
Does this need to be a list of characters? Can't it be a string?

Reply:
I put it to print the path being generated when ArtifactsHandler invokes generate_path() and it was that. Also the log that appears when download is executed, shows this as the path, I got a little confuse about it actually.

Reply:
Ah! Wait, I think I got it

Reply:
That was because I didn't add a parent_dir argument in ArtifactsHandler()